### ISSUE TITLE
ibm power s824 192 threads,P1 tabl2 need 2000sec,please help meibm power s824 192 threads,P1 tabl2 need 2000sec,please help me

### ISSUE THREAD

**Issue by yysslin** [madMAx43v3r/chia-plotter#390]
ibm power s824 192 threads,P1 tabl2 need 2000sec,please help me

**Comment by piotrasd** [comment]
post full config what is temp1, temp2 how much RAM you have etc. what OS ? filesystem ?

**Comment by yysslin** [comment]
ubunbu20.04 tmp1 sas 4t tmp2 ext3 120gb ram 
use 190 threads 128 buckets P1 T2 need 300 sec
use 16 threads 128 buckers P1 T2 need 1800 sec

**Comment by piotrasd** [comment]
my experience is with bucket 256 I have better time, second - max I can run 48 threads (24 cores)
but I'm getting final higher than I run just with 12 - basically what I tested 12 is best for me. 
NVME + RAM disk - 105GB (as 128GB total ram ) sometimes process after 5-6 cycles was killed :( but I created swap file.
and now my final time is about 1550sec

**Comment by piotrasd** [comment]
> ubunbu20.04 tmp1 sas 4t ext3 tmp2 120gb ram
> use 190 threads 128 buckets P1 T2 need 300 sec
> use 16 threads 128 buckets P1 T2 need 1800 sec

maybe here SAS drivers are slowing down process, another thing when I discovered. when I added one more DIMM 64GB all RAM start working in single mode (not dual channel) then again even with bigger amount of RAM I was getting much worst timing

**Comment by whitetechnologies** [comment]
> ubunbu20.04 tmp1 sas 4t ext3 tmp2 120gb ram
> use 190 threads 128 buckets P1 T2 need 300 sec
> use 16 threads 128 buckets P1 T2 need 1800 sec

You are getting P1 300 seconds with 190 threads? I don't think you need our help that is a great time! You overall time should be close to 1200 or so with that. Yes your SAS drive will def be the bottleneck at this point. You can try to get 512MB of RAM (easily possible on your server) and then do both TMP1 and TMP2 in RAM

**Comment by VertiHydro** [comment]
> > ubunbu20.04 tmp1 sas 4t ext3 tmp2 120gb ram
> > use 190 threads 128 buckets P1 T2 need 300 sec
> > use 16 threads 128 buckets P1 T2 need 1800 sec
> 
> You are getting P1 300 seconds with 190 threads? I don't think you need our help that is a great time! You overall time should be close to 1200 or so with that. Yes your SAS drive will def be the bottleneck at this point. You can try to get 512MB of RAM (easily possible on your server) and then do both TMP1 and TMP2 in RAM

*512GB

**Comment by VertiHydro** [comment]
> > ubunbu20.04 tmp1 sas 4t ext3 tmp2 120gb ram
> > use 190 threads 128 buckets P1 T2 need 300 sec
> > use 16 threads 128 buckets P1 T2 need 1800 sec
> 
> You are getting P1 300 seconds with 190 threads? I don't think you need our help that is a great time! You overall time should be close to 1200 or so with that. Yes your SAS drive will def be the bottleneck at this point. You can try to get 512MB of RAM (easily possible on your server) and then do both TMP1 and TMP2 in RAM

Actually I think he might be saying that Table 2 by itself takes 300 sec... but yes that's got to be a drive bottleneck

**Comment by edamame43** [comment]
Which filesystem is best for SAS drives? ext2 or ext4?

**Comment by gryan315** [comment]
In my experience, xfs is the fastest FS for SAS drives.

**Comment by yysslin** [comment]
> 
> 
> > > ubunbu20.04 tmp1 sas 4t ext3 tmp2 120gb ram
> > > use 190 threads 128 buckets P1 T2 need 300 sec
> > > use 16 threads 128 buckets P1 T2 need 1800 sec
> > 
> > 
> > You are getting P1 300 seconds with 190 threads? I don't think you need our help that is a great time! You overall time should be close to 1200 or so with that. Yes your SAS drive will def be the bottleneck at this point. You can try to get 512MB of RAM (easily possible on your server) and then do both TMP1 and TMP2 in RAM
> 
> Actually I think he might be saying that Table 2 by itself takes 300 sec... but yes that's got to be a drive bottleneck

you are right, use 192 threads  overall time need 3500 sec. I want to use 16 threads to concurrent 11 tasks.
